packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx

```
@@ -67,12 +68,17 @@ export const SettingsDevelopersApiKeyDetail = () => {
   });
 
   const deleteIntegration = async (redirect = true) => {
-    await updateApiKey?.({
-      idToUpdate: apiKeyId,
-      updateOneRecordInput: { revokedAt: DateTime.now().toString() },
-    });
-    if (redirect) {
-      navigate('/settings/developers');
+    setIsLoading(true);
+    try {
+      await updateApiKey?.({
+        idToUpdate: apiKeyId,
+        updateOneRecordInput: { revokedAt: DateTime.now().toString() },
+      });
+      if (redirect) {
+        navigate('/settings/developers');
+      }
+    } finally {
```

A catch would avoid a silent error ? You can use a snackbar.